vendor/bat-native-ledger/src/bat/ledger/internal/database/database_recurring_tip.cc

```diff
@@ -66,18 +69,139 @@ void DatabaseRecurringTip::InsertOrUpdate(
   ledger_->RunDBTransaction(std::move(transaction), transaction_callback);
 }
 
+void DatabaseRecurringTip::InsertOrUpdate(
+    const std::string& publisher_id,
+    double amount,
+    base::OnceCallback<void(bool)> callback) {
+  if (publisher_id.empty()) {
+    BLOG(1, "Publisher ID is empty");
+    std::move(callback).Run(false);
+    return;
+  }
+
+  if (amount <= 0) {
+    BLOG(1, "Invalid contribution amount");
+    std::move(callback).Run(false);
+    return;
+  }
+
+  auto transaction = mojom::DBTransaction::New();
+
+  const std::string query = base::StringPrintf(
```

Comment:
Can be non-`const`, and `std::move()`d below.

Reply:
Updated to use a temp.